tests/Sulu/Component/Rest/ListBuilder/Metadata/FieldDescriptorFactoryTest.php

```diff
@@ -0,0 +1,201 @@
+
+        $parameterBag = $this->prophesize(ParameterBagInterface::class);
+        $parameterBag->resolveValue('%sulu.model.contact.class%')->willReturn('SuluContactBundle:Contact');
+        $parameterBag->resolveValue('%sulu.model.contact.class%.avatar')->willReturn(
+            'SuluContactBundle:Contact.avatar'
+        );
+        $parameterBag->resolveValue('%sulu.model.contact.class%.contactAddresses')->willReturn(
+            'SuluContactBundle:Contact.contactAddresses'
+        );
+        $parameterBag->resolveValue(Argument::any())->willReturnArgument(0);
+
+        $configCache = $this->prophesize(ConfigCache::class);
+        $configCache->isFresh()->willReturn(false);
+        $configCache->write(Argument::any(), null)->willReturn(null);
+
+        $chain = [
+            new MetadataProvider(
+                new MetadataFactory(new DoctrineXmlDriver($locator->reveal(), $parameterBag->reveal()))
+            ),
+            new MetadataProvider(new MetadataFactory(new GeneralXmlDriver($locator->reveal()))),
+        ];
+        $provider = new ChainProvider($chain);
+        $factory = new FieldDescriptorFactory($provider, $configCache->reveal());
+        $fieldDescriptor = $factory->getFieldDescriptorForClass(\stdClass::class);
+
+        self::assertEquals(['id', 'firstName', 'lastName', 'avatar', 'fullName', 'city'], array_keys($fieldDescriptor));
+
+        $this->fieldDescriptorTest($fieldDescriptor['id'], 'id', 'public.id', true, false, 'integer');
+        $this->fieldDescriptorTest(
+            $fieldDescriptor['firstName'],
+            'firstName',
+            'contact.contacts.firstName',
+            false,
+            true
+        );
+        $this->fieldDescriptorTest($fieldDescriptor['lastName'], 'lastName', 'contact.contacts.lastName', false, true);
+        $this->fieldDescriptorTest(
+            $fieldDescriptor['avatar'],
+            'avatar',
+            'public.avatar',
+            false,
+            true,
+            'thumbnails',
+            '',
+            '',
+            false
+        );
+        $this->fieldDescriptorTest(
+            $fieldDescriptor['fullName'],
+            'fullName',
+            'public.name',
+            true,
+            false,
+            'string',
+            '100px',
+            '50px',
+            false,
+            false,
+            'test-class'
+        );
+        $this->fieldDescriptorTest($fieldDescriptor['city'], 'city', 'contact.address.city', false, true);
+
+        self::assertInstanceOf(DoctrineConcatenationFieldDescriptor::class, $fieldDescriptor['fullName']);
+        self::assertEquals(
+            'CONCAT(SuluContactBundle:Contact.firstName, CONCAT(\' \', SuluContactBundle:Contact.lastName))',
+            $fieldDescriptor['fullName']->getSelect()
+        );
+    }
+
+    public function testGetFieldDescriptorForClassMinimal()
+    {
+        $locator = $this->prophesize(FileLocatorInterface::class);
```

Comment:
Prophets should be declared in the `setUp` if they are generally used in the test class. This saves space and makes the tests more readable.